x-pack/platform/packages/shared/onechat/onechat-common/tools/types/esql.ts

```diff
@@ -34,12 +44,33 @@ export interface EsqlToolParam {
    * Description of the parameter's purpose or expected values.
    */
   description: string;
+}
+
+export interface EsqlToolParamRequired extends EsqlToolParamBase {
   /**
    * Whether the parameter is optional.
    */
-  optional?: boolean;
+  optional?: false;
+  /**
+   * Default value is not allowed for required parameters.
+   */
+  defaultValue?: never;
 }
 
+export interface EsqlToolParamOptional extends EsqlToolParamBase {
+  /**
+   * Whether the parameter is optional.
+   */
+  optional: true;
+  /**
+   * Default value for the parameter when it's optional and not provided.
+   * Must be compatible with the parameter's type (see EsqlToolParamValue).
+   */
+  defaultValue?: EsqlToolParamValue;
+}
+
+export type EsqlToolParam = EsqlToolParamRequired | EsqlToolParamOptional;
```

Comment:
I'm the first to abuse generics and type composition and all, and yet for that very specific case I would have go just with a simple

```
defaultValue?: EsqlToolParamValue;
```

What you did is fine, and structurally better - but I suspect it will be quite bothersome when working on the UI side, as you might be forced to type guard / type check everywhere?

Reply:
I'd need to test it on the UI implementation to see how bothersome it would be, but I do agree with you that we can take the more simple approach here, I'll revert this for now 😄